6105 make docket parser for SCOTUS html

albertisfu opened the pull request.

albertisfu commented:
I've applied the suggested tweak.

Basically, the title in an address is parsed until a line with a street number is found.

e.g:

1.- 
```
Bradley Arant Boult Cummings LL P<br>
1600 Division Street<br>
Suite 700<br>
Nashville, TN 37203
```

**Title:** Bradley Arant Boult Cummings LL P

**Address:** 
1600 Division Street
Suite 700
Nashville, TN 37203


2.- 
```
Acting Solicitor General
<br>
United States Department of Justice
<br>
950 Pennsylvania Avenue, N.W.
<br>
Washington, DC 20530-0001
```

**Title:** Acting Solicitor General, United States Department of Justice

**Address:** 
950 Pennsylvania Avenue, N.W.
Washington, DC 20530-0001


3.- 
```
Johnson State Prison, D-2-201-B
<br>
P.O. Box 344
<br>
Wrightsville, GA 31096-0344
```


**Title:** None

**Address:** 
Johnson State Prison, D-2-201-B
P.O. Box 344
Wrightsville, GA 31096-0344

In this case, the distinction between title and address is unclear. `Johnson State Prison, D-2-201-B` appears to be part of the address since it contains a number. Unfortunately, this docket doesn’t have a JSON version to confirm how the title should be parsed.

I also reviewed other dockets with both HTML and JSON versions and found that the title is not always clearly defined.

For instance:
https://www.supremecourt.gov/RSS/Cases/JSON/17-5574.json
<img width="673" height="590" alt="Screenshot 2025-08-28 at 11 41 53 a m" src="https://github.com/user-attachments/assets/d2bf84f8-231f-4776-8c05-5d1404e2ef66" />

Here, the title is null, but the address includes what would otherwise be considered the “title” in other cases.

Another example:

https://www.supremecourt.gov/RSS/Cases/JSON/21-1397.json
<img width="663" height="489" alt="Screenshot 2025-08-28 at 11 46 46 a m" src="https://github.com/user-attachments/assets/5842489f-0d77-4185-b555-5994fbe9340e" />

Here, the “Organization” name is used as the title.

So it seems reasonable to capture everything before the first line containing a street number as the title, since the original JSON dockets are not consistent either?

If this approach seems acceptable, I’ll apply the same tweak to the HTM parser.

The alternative would be to always merge the title into the address field.

albertisfu commented:
Great! I've encapsulated the new approach for parsing the address into a couple of methods so they can be reused in: https://github.com/freelawproject/juriscraper/pull/1557

This is ready for review.

ERosendo reviewed (approved):
Looks great overall. Just a quick question about the examples.

albertisfu commented:
Thanks @ERosendo for your review! I’ve addressed your comment and, in addition, added support for parsing a missing field `Questions Presented` in both the JSON and HTML versions, as Rachel suggested it would be a good idea to parse it as well.

ERosendo reviewed (approved):
LGTM :shipit:
